src/test/statistics/chunk_statistics/range_filter_test.cpp

```diff
@@ -0,0 +1,163 @@
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "utils/assert.hpp"
+
+#include "statistics/chunk_statistics/range_filter.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+template <typename T>
+class RangeFilterTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    _values = pmr_vector<T>{-1000, 2, 3, 4, 7, 8, 10, 17, 100, 101, 102, 103, 123456};
+    _min_value = *std::min_element(std::begin(_values), std::end(_values));
+    _max_value = *std::max_element(std::begin(_values), std::end(_values));
+
+    // value in the largest gap of the test data
+    // we changing test data, ensure that value is not part of a range in ranges unless |ranges|==1)
+    _in_between = static_cast<T>(_min_value + 0.5 * (_max_value - _min_value));
+
+    _before_range = _min_value - 1;  // value smaller than the minimum
+    _after_range = _max_value + 1;   // value larger than the maximum
+  }
+
+  void test_varying_range_filter_size(int gap_count) {
+    // RangeFilter constructor takes range count, not gap count
+    auto filter = RangeFilter<T>::build_filter(_values, gap_count + 1);
+
+    for (const auto& value : _values) {
+      EXPECT_FALSE(filter->can_prune({value}, PredicateCondition::Equals));
+    }
+
+    // find $gap_count largest gaps
+    auto value_set = std::set<T>(_values.begin(), _values.end());
+    std::vector<std::pair<T, T>> begin_length_pairs;
+
+    for (auto it = value_set.begin(); it != std::prev(value_set.end()); ++it) {
+      auto begin = *it;
+      auto end = *(std::next(it));
+      begin_length_pairs.push_back(std::make_pair(begin, end - begin));
+    }
+
+    std::sort(begin_length_pairs.begin(), begin_length_pairs.end(),
+              [](auto& left, auto& right) { return left.second > right.second; });
+
+    for (int i = 0; i < gap_count && i < static_cast<int>(begin_length_pairs.size()); ++i) {
```

Comment:
Even though it is only a test, I believe we got rid of `i` and plain `int` in most places.